core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -53,17 +55,34 @@ trait AutoTopicCreationManager {
     requestContext: RequestContext
   ): Unit
 
+  def getTopicCreationErrors(
+    topicNames: Set[String]
+  ): Map[String, String]
+
+  def close(): Unit = {}
+
+}
+
+case class CachedTopicCreationError(
+  errorMessage: String,
+  time: Time
+) {
+  val timestamp: Long = time.milliseconds()
 }
 
 class DefaultAutoTopicCreationManager(
   config: KafkaConfig,
   channelManager: NodeToControllerChannelManager,
   groupCoordinator: GroupCoordinator,
   txnCoordinator: TransactionCoordinator,
-  shareCoordinator: ShareCoordinator
+  shareCoordinator: ShareCoordinator,
+  time: Time
 ) extends AutoTopicCreationManager with Logging {
 
   private val inflightTopics = Collections.newSetFromMap(new ConcurrentHashMap[String, java.lang.Boolean]())
+  private val topicCreationErrorCache = new ConcurrentHashMap[String, CachedTopicCreationError]()
+  // Use session timeout for better semantic alignment with client lifecycle
+  private val errorCacheTtlMs = config.groupCoordinatorConfig.classicGroupMaxSessionTimeoutMs.toLong
```

Comment:
We'd want to pass this in from the outside. Session timeouts can be overwritten for a specific group, so we need to use `groupConfigManager` which is present in `KafkaApis`. 

```
        Optional<GroupConfig> groupConfig = groupConfigManager.groupConfig(groupId);
        return groupConfig.map(GroupConfig::streamsSessionTimeoutMs)
            .orElse(config.groupCoordinatorConfig.streamsGroupSessionTimeoutMs());
```

I would also re-fetch this value every time we create an error message, since the session timeout may change over time. So better pass this into `createStreamsInternalTopics`.